typeinfo-less.md

```diff
@@ -0,0 +1,305 @@
+
+`unsigned` types will be ordered after `signed`:
+
+The order of `unsigned int` and `signed int` are
+```cpp
+// ------ Lowest ------
+
+signed int
+unsigned int
+
+// ------ Highest ------
+```
+
+Sort on size modifiers first, then sort on signedness.
+
+Combining these two orderings, the order of all possible integer types would be:
+```cpp
+// ------ Lowest ------
+
+short int
+unsigned short int
+int
+unsigned int
+long int
+unsigned long int
+long long int
+unsigned long long int
+
+// ------ Highest ------
+```
+
+It is worth noting that both signedness and size modifiers only exist on native
+types, and as such will only be taken into account when sorting user defined
+types. 
+
+### Ordering References and Pointers
+
+Pointers, lvalue references, and rvalue references should be ordered as 
+follows:
+
+```cpp
+// ------ Lowest ------
+using T = ...;
+
+T // unspecified
+T* // Pointer
+T& // lvalue Reference
+T&& // rvalue Reference
+
+// ------ Highest ------
+```
+
+### Type Qualifiers
+
+`const`, `volatile` and `mutable` qualifiers should be ordered as follows:
+
+Volatile modifiers would be ordered with `unspecified < volatile`.
+
+Constantness modifiers would be ordered: `unspecified < const < mutable`
+
+Order first based on volatileness, then on constantness.
+
+For example:
+
+```cpp
+// ------ Lowest ------
+
+using T = ...;
+
+T
+const T
+mutable T
+const volatile T
+mutable volatile T
+
+// ------ Highest ------
+```
+
+### Non Templated User Defined Types
+
+Simple User Defined Types can be defines as either a `struct` or a `class` that
+has been declared without a template parameter. As far as ordering is concerned,
+structs and classes should be treated the same.
+
+User defined types will be greater than native types.
+
+Ordering these is simple, since we cannot redeclare two types with the same
+name, we'll just order simple user defined types alphabetically. 
+
+```cpp
+struct Apple {};
+class Banana {};
+struct Carrot {};
+
+```
+
+Would be ordered as `Apple < Banana < Carrot`.
+
+References and pointers to, as well as type qualifiers on, will behave the same
+way on user defined types as they do on native types. For example, given:
+
+```cpp
+Apple
+Carrot
+Apple&
+Banana const&
+Carrot&&
+```
+
+Would be ordered `Apple < Apple& < Banana const&  < Carrot < Carrot&&`
+
+### namespaces
+
+Namespaces will also be ordered alphabetically.
+
+Given:
+```cpp
+namespace foo {
+  struct i;
+}
+
+namespace bar {
+  struct i;
+}
+```
+
+The order of the two structs w/ type `i` types would be `bar::i < foo::i`
+
+Combining the above, we can define an order for every possible combination of
+type, qualifier, and pointer/reference. 
+
+1. Order alphabetically by namespace and type. Native types come before user defined types.
+2. Order by pointer or reference. 
+3. Order by type qualifier.
+
+### Type Aliases 
+
+The ordering of type aliases is to be the same as the type they are aliased to.
```

Comment:
This should really say "type aliases are not types, and we need not concern ourselves with them in this paper."